### PR TITLE
Blocking Command: add blocked error management + refactor

### PR DESCRIPTION
**Pull request by ramabit**
Documentation: https://xmpp.org/extensions/xep-0191.html#block

### CODE REVIEW

File: smack-extensions/src/main/java/org/jivesoftware/smackx/blocking/element/BlockedErrorExtension.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Override
+    public CharSequence toXML() {
+        XmlStringBuilder xml = new XmlStringBuilder(this);
+        xml.closeEmptyElement();
+        return xml;
+    }
+
+    public static BlockedErrorExtension from(Message message) {
+        return message.getError().getExtension(ELEMENT, NAMESPACE);
```
Comment (Flowdalic) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/blocking/element/BlockedErrorExtension.java]: Shouldn't this check for null?

File: smack-extensions/src/main/java/org/jivesoftware/smackx/blocking/element/BlockedErrorExtension.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    public static boolean isInside(Message message) {
+        return message.getError().getExtension(ELEMENT, NAMESPACE) != null;
```
Comment (Flowdalic) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/blocking/element/BlockedErrorExtension.java]: As above: check for null?

File: smack-extensions/src/main/java/org/jivesoftware/smackx/blocking/element/BlockedErrorExtension.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        XmlStringBuilder xml = new XmlStringBuilder(this);
+        xml.closeEmptyElement();
+        return xml;
+    }
+
+    public static BlockedErrorExtension from(Message message) {
+        return message.getError().getExtension(ELEMENT, NAMESPACE);
+    }
+
+    public static boolean isInside(Message message) {
```
Comment (Flowdalic) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/blocking/element/BlockedErrorExtension.java]: Some javadoc explaining what it means if `isInside` returns true would be nice.

**Comment by ramabit** [comment]
@Flowdalic please check the new changes :)

File: smack-extensions/src/main/java/org/jivesoftware/smackx/blocking/element/BlockedErrorExtension.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    public static BlockedErrorExtension from(Message message) {
+        XMPPError error = message.getError();
+        if (error == null) {
+            return null;
+        }
+        return error.getExtension(ELEMENT, NAMESPACE);
+    }
+
+    /**
+     * Check if a Blocked Error Extension is inside a message.
```
Comment (Flowdalic) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/blocking/element/BlockedErrorExtension.java]: But what does it mean if there is a blocked error extension inside the message?

Comment (ramabit) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/blocking/element/BlockedErrorExtension.java]: if the message contains an XMPPError inside and that contains an extension with the ELEMENT and NAMESPACE of BlockedErrorExtension

**Comment by ramabit** [comment]
@Flowdalic please check this

File: smack-extensions/src/main/java/org/jivesoftware/smackx/blocking/BlockingCommandManager.java [hunk 5]
```diff
@@ -163,14 +163,12 @@ public boolean isSupportedByServer()
     public List<Jid> getBlockList()
```
Comment (Flowdalic) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/blocking/BlockingCommandManager.java]: Why is a change to this method required?

Comment (ramabit) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/blocking/BlockingCommandManager.java]: @Flowdalic It is a refactor, look that:
`return Collections.unmodifiableList(blockListCached);`
is twice there without necessity

**Comment by Flowdalic** [comment]
Related Smack issue: https://issues.igniterealtime.org/browse/SMACK-731

**Comment by Flowdalic** [comment]
Please squash and add a nice commit message

**Comment by ramabit** [comment]
@Flowdalic done

**Comment by Flowdalic** [comment]
Cherry picked as https://github.com/igniterealtime/Smack/commit/018a7dd93a12c5f2e1f8a24b207b86ab108ff993

**Pull request closed without merging**